list/list.js

```diff
@@ -1445,7 +1453,7 @@ Object.defineProperty(DefineList.prototype, "length", {
 			return;
 		}
 
-		if (newVal === this._length) {
+		if (newVal == null || isNaN(+newVal) || newVal === this._length) {
```

Comment:
Some comments explaining these cases would be helpful.

Reply:
Done